### PR TITLE
Update events design

### PR DESCRIPTION
**Pull request by joeldudley**
### Why is this change needed?

We've made some changes to the design over time. This update covers some of these, and cleans up the document a bit for readability.

### What changes were made as part of this PR:

- Provide a high level list of the changes made
- List key areas for the reviewer 

### :rotating_light: Definition of done :rotating_light:
- [ ] Unit tests added to cover new or changed functionality 
- [ ] Docs pages updated to cover new or changed functionality
- [ ] [Changelog.md](https://github.com/obscuronet/go-obscuro/blob/main/docs/testnet/changelog.md) updated

### CODE REVIEW

File: design/Events_design.md [hunk 1]
```diff
@@ -9,64 +9,11 @@ It covers two aspects:
 - the visibility rules for events.
 - technical implementation details.
 
-## Ethereum Events Design
```
Comment (joeldudley) [inline design/Events_design.md:12]: Moved to an appendix.

File: design/Events_design.md [hunk 2]
```diff
@@ -91,109 +39,7 @@ If we were designing events from scratch, with privacy in mind, we could add met
 able to view an event. Since we're trying to maintain the API of Ethereum unchanged, we'll try to infer this information
 from the existing information available in the event, and to also allow the developers to achieve the desired outcome.
 
-Let's analyse a couple of events from ERC20 and Uniswap, grouped by whether they contain address fields.
```
Comment (joeldudley) [inline design/Events_design.md:94]: Moved to an appendix.

File: design/Events_design.md [hunk 3]
```diff
@@ -205,11 +51,17 @@ The implicit rules we propose are:
...
+
+* The event contains addresses, but they are contract addresses, as in the ``PoolCreated`` event from above
+* The event contains a topics that looks like an address, but is not
+
+To handle this, for each potential account address in the topics, the VM must check:
 
-During the evaluation phase, the VM must check each address field for the ``getCode`` function, to determine what type of address it is.  
-If at least one of the addresses is not a smart contract, then the event will fall under rule 1.
+* If there is associated contract code, using the `GetCode` function
+* If it has a zero nonce, using the `GetNonce` function (only account and smart contract addresses have non-zero nonces)
```
Comment (joeldudley) [inline design/Events_design.md:62]: Documented the new rule about nonces here.

File: design/Events_design.md [hunk 4]
```diff
@@ -296,59 +148,33 @@ authenticated, so an attacker who discovered the subscription ID could request i
...
+1. Route the new subscription request to the enclave and create a new Geth `rpc.Subscription` to return to the client
+2. Extract the logs upon each block ingestion and route them to the corresponding client subscription
 
-##### Client subscription
+Since the log subscription request is encrypted by the client and can only be decrypted on the enclave, the host 
+forwards it blindly, and cannot learn anything about the contents of the subscription. However, it does generate a 
+fresh ID for the subscription, which it also forwards to the enclave. The enclave reuses this ID when sending back 
+logs for that subscription in the block submission response.
 
-For (2), we reuse Geth's `PublicFilterAPI`, passing in a custom `Backend` object. The `Backend` object has a
```
Comment (joeldudley) [inline design/Events_design.md:331]: Was able to massively simplify this in a redesign, so rewrote this section.

**Review by BedrockSquirrel — APPROVED** [review]
LGTM

**Pull request merged**